### DISCUSSION TITLE
Better FFMPEG settings for Amcrest AD110 doorbell?

### DISCUSSION THREAD

**Discussion by jonadis** [blakeblackshear/frigate#2934] [Ask A Question]
I have an [Amcrest AD110](https://amcrest.com/2mp-wifi-camera-doorbell-ad110.html) doorbell and it works pretty well with Frigate overall, but it is rather noisy in the logs.  I'm wondering if there's perhaps some additional flags I could add to ffmpeg to improve the compatibility with this device.  Within Frigate, the camera/doorbell is quite usable, I get events as expected and I can open the live stream just like all the other cameras.  However, I see these logs some times every minute and some times there's a few hours without any issues.  

```
[2022-03-09 14:58:57] watchdog.frontdoor             INFO    : Terminating the existing ffmpeg process...
[2022-03-09 14:58:57] watchdog.frontdoor             INFO    : Waiting for ffmpeg to exit gracefully...
[2022-03-09 14:59:07] watchdog.frontdoor             ERROR   : Ffmpeg process crashed unexpectedly for frontdoor.
[2022-03-09 14:59:07] watchdog.frontdoor             ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
[2022-03-09 14:59:07] ffmpeg.frontdoor.detect        ERROR   : [h264 @ 0x55adbbec3f40] Invalid level prefix
[2022-03-09 14:59:07] ffmpeg.frontdoor.detect        ERROR   : [h264 @ 0x55adbbec3f40] error while decoding MB 29 20
[2022-03-09 14:59:07] ffmpeg.frontdoor.detect        ERROR   : Finishing stream 0:0 without any data written to it.
[2022-03-09 14:59:07] ffmpeg.frontdoor.detect        ERROR   : [swscaler @ 0x55adbc0890c0] deprecated pixel format used, make sure you did set range correctly
[2022-03-09 14:59:07] ffmpeg.frontdoor.detect        ERROR   : Output file is empty, nothing was encoded (check -ss / -t / -frames parameters if used)
[2022-03-09 14:59:27] watchdog.frontdoor             INFO    : No frames received from frontdoor in 20 seconds. Exiting ffmpeg...
[2022-03-09 14:59:27] watchdog.frontdoor             INFO    : Waiting for ffmpeg to exit gracefully...
[2022-03-09 14:59:28] frigate.video                  ERROR   : frontdoor: Unable to read frames from ffmpeg process.
[2022-03-09 14:59:28] frigate.video                  ERROR   : frontdoor: Unable to read frames from ffmpeg process.
```

My config is pretty straightforward so far:
```
  frontdoor:
    ffmpeg:
      inputs:
        - path: rtsp://admin:_redacted_@10.0.71.131:554/cam/realmonitor?channel=1&subtype=0&authbasic=64
          roles:
            - record
            - rtmp
        - path: rtsp://admin:_redacted_@10.0.71.131:554/cam/realmonitor?channel=1&subtype=1&authbasic=64
          roles:
            - detect
    detect:
      width: 640
      height: 480
      fps: 5
    motion:
      mask:
        - 0,128,640,128,640,0,0,0
```
Any advise is appreciated.

**Comment by PrplHaz4** [comment]
I see these with the AD110 too - per responses in other forums I think they are mostly related to wifi strength and power management, but would definitely be interested in some ffmpeg args that make the stream a little more fault-tolerant (without adding latency?)...

**Comment by NickM-27** [comment]
I have the AD 410 and interesting to see as I don't have any errors coming from it. Not sure what differences between the streams? I know the AD 110 has a worse WiFi radio so perhaps it is a signal issue? 🤔

**Comment by jmswindle** [comment]
I have the AD110 as well and was constantly having stability issues. I didn't realize my transformer was only 16VAC 10VA - so I replaced it with a 16VAC 30VA one and have had much better reliability. Still not 100% though, unfortunately, makes me wary to invest in any more amcrest products.

**Reply by NickM-27** [reply]
The AD110 was an unfortunate blunder on their part (I've seen many meh reviews of it) but their recent offerings have been much better in my experience.

**Comment by sharbich** [comment]
Hello,
I can't get my AD110 to run. Here are my error messages

`2023-04-11 09:16:53.719093974  [2023-04-11 11:16:53] frigate.comms.mqtt
ERROR   : Unable to publish to stats: client is not connected`

my config

`root@dsme01:~# cat /usr/share/hassio/homeassistant/frigate.yml 
mqtt:
  host: homeassitant.intern.example.com
cameras:
  back:
    ffmpeg:
      inputs:
        - path: rtsp://admin:Marlies1234@192.168.30.249:554/cam/realmonitor?channel=1&subtype=0&authbasic=64
          roles:
            - record
            - rtmp
        - path: rtsp://admin:Marlies1234@192.168.30.249:554/cam/realmonitor?channel=1&subtype=1&authbasic=64
          roles:
            - detect
    detect:
      width: 640
      height: 480
      fps: 5
    motion:
      mask:
        - 0,128,640,128,640,0,0,0`

 Can you help me?

**Reply by dison4linux** [reply]
Those don't look like error messages related to the doorbell connection, they look like MQTT error messages.
Can you look again for error messages?
Can you open that RTSP stream with VLC?

**Reply by sharbich** [reply]
Hello,
i used an incorrect password when logging into mqtt broker. It works now.
But I still have a problem. How do I get an audio connection? I do not hear anything. Neither on the microphones of the doorbell nor on the client.

Greetings from Stefan Harbich

**Reply by kdill00** [reply]
Is there Audio in recorded clips?  If you go to the system tab, do an ffprobe on the doorbell camera, does it show a second “stream” for audio?  Jsmpeg live view will not play audio. You will need to setup webrtc if you want audio in live view. If the stream for audio is enabled and exists then it should be just selecting an output preset that will work for your camera. Please post config if you have further issues

**Comment by Vendo232** [comment]
here is my setup, works great.

go2rtc:
  streams:
    front_door_bell_high:
      - rtsp://admin:password@192.168.1.95:554/cam/realmonitor?channel=1&subtype=0
    front_door_bell_low:
      - rtsp://admin:password@192.168.1.95:554/cam/realmonitor?channel=1&subtype=1

  front_door_bell:
    enabled: True
    ffmpeg:
      inputs:
        - path: rtsp://127.0.0.1:8554/front_door_bell_low
          input_args: preset-rtsp-restream
          roles:
            - detect
        - path: rtsp://127.0.0.1:8554/front_door_bell_high
          input_args: preset-rtsp-restream
          roles:
            - record
